api/messages.py

```diff
@@ -0,0 +1,207 @@
+from ..harness import Harness
+from ..request import Request, Response, TransactionError
+from ..interface.defs import Endpoint, Method
+from .. import log
+from .generic import GenericResponse, GenericTransaction
+
+class Threads(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.entries = self.response.body["entries"]
+        self.totalCount = self.response.body["totalCount"]
+        self.nextPage = self.response.body["nextPage"]
+
+class GetThreadsWithOffsetAndLimit(GenericTransaction):
+    '''
+    Retrieve threads with offset and limit
+    '''
+    def __init__(self, offset: int, limit: int):
+        self.request = Request(Endpoint.MESSAGES, Method.GET, {"category":"thread", "offset": offset, "limit": limit})
+
+    def setResponse(self, response: Response):
+        self.response = Threads(response)
+
+class MarkThreadAsUnread(GenericTransaction):
+    '''
+    Mark thread as read/unread
+    '''
+    def __init__(self, threadID: int, isUnread: int):
+        self.request = Request(Endpoint.MESSAGES, Method.GET, {"category":"thread", "threadID": threadID, "isUnread": isUnread})
+
+    def setResponse(self, response: Response):
+        self.response = response
+
+'''
+Messages
+'''
+
+class MessagesCount(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.count = self.response.body["count"]
+
+class Messages(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.messages = self.response.body["entries"]
+        self.totalCount = self.response.body["totalCount"]
+
+class MessageByIdEntry(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.message = self.response.body
+
+class AddedMessage(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.message = self.response.body
+
+class DeletedMessage(GenericResponse):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
Not sure if this class is needed.
Probably GenericResponse could be used directly instead.

Reply:
Right, this is redundant.